mlir/lib/CAPI/Dialect/MIGraphX.cpp

```diff
@@ -136,6 +137,7 @@ MLIR_CAPI_EXPORTED bool mlirGetBinary(MlirModule module, size_t *size,
 MLIR_CAPI_EXPORTED
 void mlirMIGraphXAddHighLevelPipeline(MlirPassManager pm) {
   auto passMan = unwrap(pm);
+  mlir::registerRocMLIROptions();
```

Comment:
Register the options here, which is called from MIGraphX mlir.cpp's run_high_level_pipeline().

Reply:
I'm not sure we should be doing this, since you shouldn't be re-parsing the command line multiple times

Reply:
That is, `registerMLIRCLOptions()` should be called exactly once, which is why we want MIGraphX to do it - just like they need to call dialect registration once

(Like, it might be *safe* to call it repeatedly, but ... why)

Reply:
Well, color me confused.  Way back at the beginning, I tried doing everything at the one-time-registration point in MIGraphX, and it wouldn't do print-ir, which led me to doing it in the add-pipeline functions.

Following your comments, I decided to trace that problem more thoroughly, and now it works the way I want it to -- calling the registration function at the same time we register passes and dlalects does in fact lead to recognising --mlir-print-ir-after-all.

I'm going to sit on this for a bit, wipe out all my test installations and caches, and try again from scratch, to make sure that's what's really happening.  Either I'll be able to do it one time, where the other registrations happen, or I'll understand better why I can't.

BTW, registration for command-line options isn't parsing, it's just creation and initialisation.  It's probably innocuous to repeat it, but it's cleaner not to.

Reply:
So let's get rid of this bit?

Reply:
Yes.  The cheating way is to replace the call to mlir::registerMLIRCLOptions() in mlirRegisterRocMLIRPasses() with a call to mlirRegisterRocMLIROptions() (or whatever we rename it too).  In other words, sneak it into a function that's already called only once.  The ostensibly cleaner way is to call mlirRegisterRocMLIROptions() from get_dialect_registry() in mlir.cpp alongside the other registration functions.  Assuming it continues to work.

Reply:
Yeah, I'd rather call it in the dialect registry getter or something else that runs as part of library initialization